src/lambda/eventConsumer/userEvents/userEventConsumer.ts

```diff
@@ -0,0 +1,72 @@
+import { UserEventHandler } from '../../snowplow/userEventHandler';
+import {
+  EventTypeString,
+  UserEventPayloadSnowplow,
+} from '../../snowplow/types';
+import { SQSRecord } from 'aws-lambda';
+
+export type UserEventPayload = {
+  userId: string;
+  email: string;
+  isPremium?: boolean;
+  apiId: string;
+  hashedId?: string;
+  guid?: number;
+  hashedGuid?: string;
+  name?: string;
+  isNative?: boolean;
+  isTrusted?: boolean;
+  clientVersion?: string;
+  language?: string;
+  snowplowDomainUserId?: string;
+  ipAddress?: string;
+  userAgent?: string;
+};
+
+//detail-type in event rule
+//defined here:
+// https://github.com/Pocket/pocket-event-bridge/blob/f462cbb1b166d937fcd62319f88c90efc7653ebc/.aws/src/event-rules/user-api-events/eventConfig.ts#L3
+export const DetailTypeToSnowplowMap: Record<string, EventTypeString> = {
+  'account-deletion': 'ACCOUNT_DELETE',
+  'account-email-updated': 'ACCOUNT_EMAIL_UPDATED',
+};
+
+export async function userEventConsumer(record: SQSRecord) {
+  await new UserEventHandler().process(getUserEventPayload(record.body));
+}
+
+export function getUserEventPayload(eventObj: any): UserEventPayloadSnowplow {
+  eventObj = JSON.parse(eventObj);
+  const message = JSON.parse(eventObj.Message);
+  const messageBody: UserEventPayload = message['detail'];
+  const detailType = message['detail-type'];
+
+  return {
```

Comment:
since all the keys are the same and everything but isPremium is a passthrough, you might consider just filtering for null values, e.g.

```
const filteredMessage = Object.entries(message).filter(([_, val]) => val != null)
return {user: ...filteredMessage, isPremium: messageBody.isPremium ? true: false }
```

Could add an additional filter if you just want to ensure only a list of specific keys are included.

Reply:
oh i couldn't do it as the other end is flat structure, i.e event bridge payload doesn't differentiate `user` , `apiUser` wheras the snowplow does